ocaml/testsuite/tests/typing-layouts/jkinds.ml

```diff
@@ -84,6 +87,372 @@ Uncaught exception: Misc.Fatal_error
+
+type a : value mod global
+type b : any mod local = a
+[%%expect{|
+type a : value mod global
+type b = a
+|}]
+
+type a : value mod global
+type b : float32 mod local = a
+[%%expect{|
+type a : value mod global
+Line 2, characters 0-30:
+2 | type b : float32 mod local = a
+    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Error: The layout of type a is value, because
+         of the definition of a at line 1, characters 0-25.
+       But the layout of type a must be a sublayout of float32, because
+         of the definition of b at line 2, characters 0-30.
+|}]
+
+type a : value mod global unique many uncontended portable external_
+type b : value mod local shared once contended nonportable internal = a
+[%%expect{|
+type a : value mod global unique many uncontended portable external_
+type b = a
+|}]
+
+type a : value mod global unique once uncontended portable external_
+type b : value mod local shared many uncontended nonportable internal = a
+[%%expect{|
+type a : value mod global unique once uncontended portable external_
+Line 2, characters 0-73:
+2 | type b : value mod local shared many uncontended nonportable internal = a
+    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Error: The layout of type a is immediate, because
+         of the definition of a at line 1, characters 0-68.
+       But the layout of type a must be a sublayout of value, because
+         of the definition of b at line 2, characters 0-73.
+|}]
+
+(********************************************************)
+(* Test 3: Abbreviation primitives are properly defined *)
+(* CR layouts: when we have abbreviations, these tests can become less verbose *)
+
+type a : any
+type b : any mod local shared once contended nonportable internal = a
+type c : any mod local shared once contended nonportable internal
+type d : any = c
+[%%expect{|
+type a : any
+type b = a
+type c : any mod local shared once contended nonportable internal
+type d = c
+|}]
+
+type a : value
+type b : value mod local shared once contended nonportable internal = a
+type c : value mod local shared once contended nonportable internal
+type d : value = c
+[%%expect{|
+type a : value
+type b = a
+type c : value mod local shared once contended nonportable internal
+type d = c
+|}]
+
+type a : void
+type b : void mod local shared once contended nonportable internal = a
+type c : void mod local shared once contended nonportable internal
+type d : void = c
+[%%expect{|
+Line 1, characters 9-13:
+1 | type a : void
+             ^^^^
+Error: Layout void is more experimental than allowed by the enabled layouts extension.
+       You must enable -extension layouts_alpha to use this feature.
+|}]
+
+type a : immediate
+type b : value mod global unique many uncontended portable external_ = a
+type c : value mod global unique many uncontended portable external_
+type d : immediate = c
+[%%expect{|
+type a : immediate
+type b = a
+type c : value mod global unique many uncontended portable external_
```

Comment:
I'm surprised at this output. It's fine -- it's what the user wrote. But I somehow expected to see `immediate` here.

Reply:
There's a comment in `ocaml/typing/printtyp.ml` that discusses jkind printing. I attempted to keep this comment true with my changes. This case refers to `C1.1` in that comment, which dictates that the user-written jkind is printed for abstract types with no manifest.

Reply:
To clarify though, I wouldn't be sad if that logic changes

Reply:
Makes sense. I wrote that comment some time ago, just feeling my way through it all. At that point, if we printed an annotation, what we printed was unambiguous. Now, the decision of whether to print and what to print are different. That is, if someone writes `type t : value`, should we print the annotation? I was warming up to say "yes"... but actually I think we shouldn't. We should feel free to update the description and the code as time evolves. Maybe even delete the comment if it becomes obvious in context or something.

Reply:
I've updated C1.1 to not print the jkind when it is equal to `value`, and updated the tests to reflect it. I am mildly concerned that a user might get confused if they write, say, `type t : value mod nonportable`, and the error message only refers to `type t`. If they included the `nonportable`, they probably didn't realize that it's a top and therefore a no-op. And if they don't realize that, not including it in the error could be confusing. But I guess if there's warnings for moding by a top, this is a non-issue.